### PR TITLE
Course maintenance: Ask for course code before deleting *or* recycling  #3302

### PR DESCRIPTION
**Pull request by carlangas159**
Course maintenance: Ask for course code before deleting *or* recycling  #3302

On recycling, ask for course code to complete recycle.
Languaje CourseCodeConfirmation

### CODE REVIEW

File: main/coursecopy/recycle_course.php [hunk 1]
```diff
@@ -55,14 +59,41 @@
...
+    if ($recycle_type == 'full_backup')
+        {
+            /* to delete, course code confirmation must be equal that current course code */
+            if ($current_course_code == $courseCodeConfirmation)
+                {
+                    $cr->recycle($recycle_type);
+                } else
+                {
+                    $messageFailCourseCode = '<p>' . get_lang('CourseRegistrationCodeIncorrect') . '</p>';
+                    $messageFailCourseCode .= '<p><a class="btn btn-primary" href="' . api_get_self() . '?' . api_get_cidreq() . '">' . get_lang('BackToPreviousPage') . '</a></p>';
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:88]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 180 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

**Comment by jmontoyaa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344&bid=18718295)

File: main/course_info/delete_course.php [hunk 2]
```diff
@@ -25,29 +25,49 @@
...
-        get_lang('No').'</a>&nbsp;<a class="btn btn-danger" href="'.api_get_self().'?delete=yes&'.api_get_cidreq().'">'.
-        get_lang('Yes').'</a></p>';
+    $message = '<h3>' . get_lang('Course') . ' : ' . $current_course_name . ' (' . $current_course_code . ') </h3>';
+    $message .= '<p>' . get_lang('ByDel') . '</p>';
+    $message .= '<p><span class="form_required">*</span>' . get_lang('CourseCodeConfirmation') . '&nbsp;<input type="text" name="course_code" id="course_code"></p>';
+
+    $message .= '<p>';
+    $message .= '<button class="btn btn-danger delete-course">' . get_lang('ValidateChanges') . '</button>';
+    $message .= '&nbsp;';
+    $message .= '<a class="btn btn-primary"href="' . api_get_path(WEB_CODE_PATH) . 'course_info/maintenance.php?' . api_get_cidreq() . '">' .
```
Comment (jmontoyaa) [inline main/course_info/delete_course.php:54]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 141 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 3]
```diff
@@ -82,18 +113,40 @@
...
     } else {
-        echo Display::return_message(get_lang('RecycleWarning'), 'warning', false);
+         echo Display::return_message(get_lang('RecycleWarning'), 'warning', false);
         $form = new FormValidator('recycle_course', 'post', api_get_self().'?'.api_get_cidreq());
         $form->addElement('header', get_lang('SelectOptionForBackup'));
         $form->addElement('radio', 'recycle_option', null, get_lang('FullRecycle'), 'full_backup');
         $form->addElement('radio', 'recycle_option', null, get_lang('LetMeSelectItems'), 'select_items');
+
+        //Confirmation input code
+        $form->addElement('label', '', '<span class="hidden course-full-delete">'.get_lang('CourseCodeConfirmation').'</span>', null, null);
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:123]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 140 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 4]
```diff
@@ -55,14 +59,41 @@
...
+                {
+                    $cr->recycle($recycle_type);
+                } else
+                {
+                    $messageFailCourseCode = '<p>' . get_lang('CourseRegistrationCodeIncorrect') . '</p>';
+                    $messageFailCourseCode .= '<p><a class="btn btn-primary" href="' . api_get_self() . '?' . api_get_cidreq() . '">' . get_lang('BackToPreviousPage') . '</a></p>';
+                    echo Display::return_message($messageFailCourseCode, 'error', false);
+                }
+
+        } elseif ($recycle_type == 'select_items')
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:92]: Codacy found an issue: [Line indented incorrectly; expected 4 spaces, found 8](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 5]
```diff
@@ -55,14 +59,41 @@
         $course = $cb->build();
     }
     $recycle_type = '';
+    $fullDelete = 0;
+    $courseCodeConfirmation ='';
+    if (isset($_POST['course_code_confirmation']))
+        {
+            $courseCodeConfirmation = $_POST['course_code_confirmation'];
+        }
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:67]: Codacy found an issue: [Line indented incorrectly; expected 4 spaces, found 8](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/course_info/delete_course.php [hunk 6]
```diff
@@ -25,29 +25,49 @@
...
+        Session::erase('_cid');
+        Session::erase('_real_cid');
+        $message = '<h2>' . get_lang('Course') . ' : ' . $current_course_name . ' (' . $current_course_code . ') </h2>';
+        $message .= get_lang('HasDel');
+        $message .= '<br /><br /><a href="../../index.php">' . get_lang('BackHome') . '</a>';
+    } else {
+        /* message if code course is incorrect */
+        $message = '<h2>' . get_lang('Course') . ' : ' . $current_course_name . ' (' . $current_course_code . ') </h2>';
+        $message .= '<p>'.get_lang('CourseRegistrationCodeIncorrect').'</p>';
+        $message .= '<p><a class="btn btn-primary" href="' . api_get_path(WEB_CODE_PATH) . 'course_info/delete_course.php?' . api_get_cidreq() . '">' . get_lang('BackToPreviousPage') . '</a>';
```
Comment (jmontoyaa) [inline main/course_info/delete_course.php:42]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 192 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 7]
```diff
@@ -55,14 +59,41 @@
...
     }
+
     $cr = new CourseRecycler($course);
-    $cr->recycle($recycle_type);
-    echo Display::return_message(get_lang('RecycleFinished'), 'confirm');
+
+    if ($recycle_type == 'full_backup')
+        {
+            /* to delete, course code confirmation must be equal that current course code */
+            if ($current_course_code == $courseCodeConfirmation)
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:82]: Codacy found an issue: [Line indented incorrectly; expected 8 spaces, found 12](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 8]
```diff
@@ -55,14 +59,41 @@
...
+                {
+                    $cr->recycle($recycle_type);
+                } else
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:85]: Codacy found an issue: [Line indented incorrectly; expected 8 spaces, found 16](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/course_info/delete_course.php [hunk 9]
```diff
@@ -25,29 +25,49 @@
...
 } else {
-    $message = '<h3>'.get_lang('Course').' : '.$current_course_name.' ('.$current_course_code.') </h3>';
-    $message .= '<p>'.get_lang('ByDel').'</p>';
-    $message .= '<p><a class="btn btn-primary" 
-        href="'.api_get_path(WEB_CODE_PATH).'course_info/maintenance.php?'.api_get_cidreq().'">'.
-        get_lang('No').'</a>&nbsp;<a class="btn btn-danger" href="'.api_get_self().'?delete=yes&'.api_get_cidreq().'">'.
-        get_lang('Yes').'</a></p>';
+    $message = '<h3>' . get_lang('Course') . ' : ' . $current_course_name . ' (' . $current_course_code . ') </h3>';
+    $message .= '<p>' . get_lang('ByDel') . '</p>';
+    $message .= '<p><span class="form_required">*</span>' . get_lang('CourseCodeConfirmation') . '&nbsp;<input type="text" name="course_code" id="course_code"></p>';
```
Comment (jmontoyaa) [inline main/course_info/delete_course.php:49]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 165 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

**Comment by jmontoyaa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344&bid=18718295)

File: main/coursecopy/recycle_course.php [hunk 10]
```diff
@@ -55,14 +59,42 @@
...
     }
+
     $cr = new CourseRecycler($course);
-    $cr->recycle($recycle_type);
-    echo Display::return_message(get_lang('RecycleFinished'), 'confirm');
+
+    if ($recycle_type == 'full_backup')
+        {
+            /* to delete, course code confirmation must be equal that current course code */
+            if ($current_course_code == $courseCodeConfirmation)
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:82]: Codacy found an issue: [Line indented incorrectly; expected 8 spaces, found 12](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 11]
```diff
@@ -55,14 +59,42 @@
...
+                    $cr->recycle($recycle_type);
+                    echo Display::return_message(get_lang('RecycleFinished'), 'confirm');
+                } else
+                {
+                    $messageFailCourseCode = '<p>' . get_lang('CourseRegistrationCodeIncorrect') . '</p>';
+                    $messageFailCourseCode .= '<p><a class="btn btn-primary" href="' . api_get_self() . '?' . api_get_cidreq() . '">' . get_lang('BackToPreviousPage') . '</a></p>';
+                    echo Display::return_message($messageFailCourseCode, 'error', false);
+                }
+
+        } elseif ($recycle_type == 'select_items')
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:93]: Codacy found an issue: [Line indented incorrectly; expected 4 spaces, found 8](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 12]
```diff
@@ -55,14 +59,42 @@
...
+        {
+            /* to delete, course code confirmation must be equal that current course code */
+            if ($current_course_code == $courseCodeConfirmation)
+                {
+                    $cr->recycle($recycle_type);
+                    echo Display::return_message(get_lang('RecycleFinished'), 'confirm');
+                } else
+                {
+                    $messageFailCourseCode = '<p>' . get_lang('CourseRegistrationCodeIncorrect') . '</p>';
+                    $messageFailCourseCode .= '<p><a class="btn btn-primary" href="' . api_get_self() . '?' . api_get_cidreq() . '">' . get_lang('BackToPreviousPage') . '</a></p>';
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:89]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 180 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/course_info/delete_course.php (hunk 6 above)
Comment (jmontoyaa) [inline main/course_info/delete_course.php:42]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 192 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 13]
```diff
@@ -55,14 +59,42 @@
...
-    echo Display::return_message(get_lang('RecycleFinished'), 'confirm');
+
+    if ($recycle_type == 'full_backup')
+        {
+            /* to delete, course code confirmation must be equal that current course code */
+            if ($current_course_code == $courseCodeConfirmation)
+                {
+                    $cr->recycle($recycle_type);
+                    echo Display::return_message(get_lang('RecycleFinished'), 'confirm');
+                } else
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:86]: Codacy found an issue: [Line indented incorrectly; expected 8 spaces, found 16](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 14]
```diff
@@ -82,18 +114,40 @@
...
-        echo Display::return_message(get_lang('RecycleWarning'), 'warning', false);
+         echo Display::return_message(get_lang('RecycleWarning'), 'warning', false);
         $form = new FormValidator('recycle_course', 'post', api_get_self().'?'.api_get_cidreq());
         $form->addElement('header', get_lang('SelectOptionForBackup'));
         $form->addElement('radio', 'recycle_option', null, get_lang('FullRecycle'), 'full_backup');
         $form->addElement('radio', 'recycle_option', null, get_lang('LetMeSelectItems'), 'select_items');
+
+        //Confirmation input code
+        $form->addElement('label', '', '<span class="hidden course-full-delete">'.get_lang('CourseCodeConfirmation').'</span>', null, null);
+        $form->addElement('input', 'course_code_confirmation', null, 'class="hidden course-full-delete"', 'course_code');
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:125]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 121 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 15]
```diff
@@ -82,18 +114,40 @@
...
     } else {
-        echo Display::return_message(get_lang('RecycleWarning'), 'warning', false);
+         echo Display::return_message(get_lang('RecycleWarning'), 'warning', false);
         $form = new FormValidator('recycle_course', 'post', api_get_self().'?'.api_get_cidreq());
         $form->addElement('header', get_lang('SelectOptionForBackup'));
         $form->addElement('radio', 'recycle_option', null, get_lang('FullRecycle'), 'full_backup');
         $form->addElement('radio', 'recycle_option', null, get_lang('LetMeSelectItems'), 'select_items');
+
+        //Confirmation input code
+        $form->addElement('label', '', '<span class="hidden course-full-delete">'.get_lang('CourseCodeConfirmation').'</span>', null, null);
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:124]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 140 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/course_info/delete_course.php (hunk 9 above)
Comment (jmontoyaa) [inline main/course_info/delete_course.php:49]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 165 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 16]
```diff
@@ -55,14 +59,42 @@
         $course = $cb->build();
     }
     $recycle_type = '';
+    $fullDelete = 0;
+    $courseCodeConfirmation ='';
+    if (isset($_POST['course_code_confirmation']))
+        {
+            $courseCodeConfirmation = $_POST['course_code_confirmation'];
+        }
```
Comment (jmontoyaa) [inline main/coursecopy/recycle_course.php:67]: Codacy found an issue: [Line indented incorrectly; expected 4 spaces, found 8](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/course_info/delete_course.php (hunk 2 above)
Comment (jmontoyaa) [inline main/course_info/delete_course.php:54]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 141 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)

File: main/coursecopy/recycle_course.php [hunk 17]
```diff
@@ -82,18 +114,40 @@
     if (!$course->has_resources()) {
         echo get_lang('NoResourcesToRecycle');
     } else {
-        echo Display::return_message(get_lang('RecycleWarning'), 'warning', false);
+         echo Display::return_message(get_lang('RecycleWarning'), 'warning', false);
```
Comment (ywarnier) [inline main/coursecopy/recycle_course.php:117]: espacio incorrecto

File: main/coursecopy/recycle_course.php (hunk 11 above)
Comment (ywarnier) [reply main/coursecopy/recycle_course.php:93]: Spacing needs to be corrected.

File: main/coursecopy/recycle_course.php (hunk 13 above)
Comment (ywarnier) [reply main/coursecopy/recycle_course.php:86]: Spacing needs to be corrected.

File: main/coursecopy/recycle_course.php (hunk 10 above)
Comment (ywarnier) [reply main/coursecopy/recycle_course.php:82]: Spacing needs to be corrected

File: main/coursecopy/recycle_course.php (hunk 16 above)
Comment (ywarnier) [reply main/coursecopy/recycle_course.php:67]: Spacing needs to be corrected

**Pull request merged**